### PR TITLE
[BEAM-2365] Use the highest pickle protocol available.

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @vikkyrk

**Comment by robertwb** [comment]
jenkins: retest this please

File: sdks/python/apache_beam/coders/coders.py [hunk 1]
```diff
@@ -365,7 +365,7 @@ def maybe_dill_dumps(o):
   # We need to use the dill pickler for objects of certain custom classes,
   # including, for example, ones that contain lambdas.
   try:
-    return pickle.dumps(o)
```
Comment (vikkyrk) [inline sdks/python/apache_beam/coders/coders.py]: Use `https://docs.python.org/2/library/pickle.html#pickle.HIGHEST_PROTOCOL` to be more clear with the intent?

**Comment by vikkyrk** [comment]
LGTM

File: sdks/python/apache_beam/coders/coders.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/coders/coders.py]: Done.

**Pull request closed without merging**